### PR TITLE
Handle Non-IDE events

### PR DESCRIPTION
**Pull request by saleem-latif**
Note: I am currently working on testing the flow. I will keep pushing the updates, but the PR is ready for initial review.

### CODE REVIEW

File: src/main/java/org/intellij/sdk/codesync/PopulateBuffer.java [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+/*
+    This class will handle non-IDE events, it will look through the files in
```
Comment (Basit-qc) [inline src/main/java/org/intellij/sdk/codesync/PopulateBuffer.java:27]: 👍 Copying this comment in other plugins code.

**Comment by Basit-qc** [comment]
Looks great overall. 👍

**Review by Basit-qc — APPROVED** [review]

File: src/main/java/org/intellij/sdk/codesync/utils/FileUtils.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            return null;
+        }
+    }
+
+    /*
+        List absolute file paths of all the files in a directory.
+
+        This is recursively list all the files containing in the given directory and all its sub-directories.
+         */
+    public static String[] listFiles(String directory) {
```
Comment (Basit-qc) [inline src/main/java/org/intellij/sdk/codesync/utils/FileUtils.java:57]: If there is no .syncigore, it syncs all the files, right?

Comment (saleem-latif) [reply src/main/java/org/intellij/sdk/codesync/utils/FileUtils.java:57]: Yes

**Pull request merged**